graphql/resolvers.go

```diff
@@ -365,7 +366,30 @@ func (r *mutationResolver) SpawnHost(ctx context.Context, spawnHostInput *SpawnH
 		options.Expiration = spawnHostInput.Expiration
 	}
 
+	if util.IsPtrSetToTrue(spawnHostInput.UseProjectSetupScript) {
+		if spawnHostInput.TaskID == nil {
+			return nil, ResourceNotFound.Send(ctx, "A valid task id must be supplied when useProjectSetupScript is set to true")
+		}
+		options.UseProjectSetupScript = *spawnHostInput.UseProjectSetupScript
+		options.TaskID = *spawnHostInput.TaskID
+	}
+
 	hc := &data.DBConnector{}
+
+	if util.IsPtrSetToTrue(spawnHostInput.SpawnHostsStartedByTask) {
+		if spawnHostInput.TaskID == nil {
+			return nil, ResourceNotFound.Send(ctx, "A valid task id must be supplied when SpawnHostsStartedByTask is set to true")
+		}
+		task, e := task.FindOneId(*spawnHostInput.TaskID)
```

Comment:
What's the purpose of using e instead of the usual "err" in this block? we do seem to treat it as a regular error so it seems a bit misleading

Reply:
Ah, usually what we do to get around that is separately declare the other variable (here, `var task *task.Task`) and then use `task, err = task.findOne()` because then you aren't overwriting err. I think it's best to use err unless you're handling the error in an unusual way, for readability.

Reply:
Thanks! Just made that change.